hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
@hyi - the resource should be immutable at the point that they agree to the publication agreement and proceed with publication.  So, regardless of how long it takes to get the DOI registered, the resource should be immutable to the user when they click "OK" or whatever and not wait until the DOI has been registered.  If the immutable flag can be used for this purpose, great, but the edit button on the page should also disappear and so they wouldn't be able to get to the edit page anyway right?